tableauserverclient/server/endpoint/workbooks_endpoint.py

```diff
@@ -291,22 +300,32 @@ def publish(
             url += '&{0}=true'.format('asJob')
 
         # Determine if chunking is required (64MB is the limit for single upload method)
-        if os.path.getsize(file_path) >= FILESIZE_LIMIT:
-            logger.info('Publishing {0} to server with chunking method (workbook over 64MB)'.format(filename))
-            upload_session_id = Fileuploads.upload_chunks(self.parent_srv, file_path)
+        if is_file_path:
+            file_size = os.path.getsize(file)
+        else:
+            file.seek(0, os.SEEK_END)
```

Comment:
I had to look this up, but I see what’s going on here — is it possible to safely use os.fstat(fd).st_size? If not, can you extract this into a little utility function named `get_file_size` or something similar?

Reply:
As I understood it, in memory binary streams such as BytesIO() don't have a file descriptor that is needed for os.fstat. I'll extract the logic to a utility function
```
from io import BytesIO()
import os

with open('file.txt', 'wb') as f:
    f.write('Hello World!'.encode())

f1 = open('file.txt')

f2 = BytesIO()
f2.write('Hello World!'.encode())

print('f1 size', os.fstat(f1.fileno()).st_size)
print('f2 size', os.fstat(f2.fileno()).st_size)

> f1 size 12
> Traceback (most recent call last):
>  File "<stdin>", line 1, in <module>
> io.UnsupportedOperation: fileno
```